File: Data Structures & Algorithms/palindromic-substrings/submission-1.java
class Solution {
    private int helper(char[] cs, int left, int right) {
        int count = 0;
        while (left >= 0 && right < cs.length && cs[left] == cs[right]) {
            count ++;
            left --;
            right ++;
        }

        return count;
    }
    public int countSubstrings(String s) {
        if (s.length() <= 1) return s.length();

        int res = 0;
        char[] cs = s.toCharArray();
        for (int i = 0; i < cs.length; i++) {
            res += helper(cs, i, i);
            res += helper(cs, i, i + 1);
        }
        
        return res;
    }
}

File: Data Structures & Algorithms/minimum-window-with-characters/submission-1.py
class Solution:
    def minWindow(self, s: str, t: str) -> str:
        res = s + t
        count = Counter(t)
        flag = len(count)
        lo, hi = 0, 0

        while hi < len(s):
            if s[hi] in count:
                count[s[hi]] -= 1
                if count[s[hi]] == 0:
                    flag -= 1
            hi += 1
            while flag == 0:
                res = s[lo: hi] if (hi - lo) < len(res) else res
                
                if s[lo] in count:
                    count[s[lo]] += 1
                    if count[s[lo]] == 1:
                        flag += 1
                lo += 1
        
        return res if res != s + t else ""